wp-content/themes/twentytwenty-child/functions.php

```diff
@@ -293,4 +299,94 @@ function update_students_settings() {
     }
 }
 
+/** 
+ *  Add custom sub menu page 
+ */
+function ajax_student_administration() {
+    add_submenu_page(
+        'student-administration',
+        'AJAX Students',
+        'AJAX Students',
+        'manage_options',
+        'ajax-student-administration',
+        'ajax_admin_callback' );
+}
+add_action( 'admin_menu', 'ajax_student_administration' );
+
+function ajax_admin_callback() {
+    update_students_settings();
+    echo '<div class="wrap">' . '<h2>AJAX Students Administration</h2>' . '</div>' . '<form method="post">';
+    settings_fields( 'ajax-student-administration' );
+    do_settings_sections( 'ajax-student-administration' );
+}
+
+/* Add custom settings to the new settings page */
+function ajax_register_settings() {
+    add_settings_section( 'students', 'Display Settings', null, 'ajax-student-administration' );
+
+    add_settings_field( 'show_country'   , 'Show country:'        , 'show_checkbox_country_ajax'  , 'ajax-student-administration', 'students' );
+    add_settings_field( 'show_address'   , 'Show address:'        , 'show_checkbox_address_ajax'  , 'ajax-student-administration', 'students' );
+    add_settings_field( 'show_birthdate' , 'Show birthdate:'      , 'show_checkbox_birthdate_ajax', 'ajax-student-administration', 'students' );
+    add_settings_field( 'show_class'     , 'Show class:'          , 'show_checkbox_class_ajax'    , 'ajax-student-administration', 'students' );
+    add_settings_field( 'show_status'    , 'Show activity status:', 'show_checkbox_status_ajax'   , 'ajax-student-administration', 'students' );
+
+    register_setting( 'ajax-student-administration', 'show_country' );
+    register_setting( 'ajax-student-administration', 'show_address' );
+    register_setting( 'ajax-student-administration', 'show_birthdate' );
+    register_setting( 'ajax-student-administration', 'show_class' );
+    register_setting( 'ajax-student-administration', 'show_status' );
+}
+add_action( 'admin_init', 'ajax_register_settings' );
+
+/* Callback for a custom setting */
+function show_checkbox_country_ajax() {
+    $options = get_option( 'show_ajax_settings' );
+    $checked = '';
+    if( $options["show_country"] == 1 ) $checked = 'checked';
+    echo '<input type="checkbox" id="show_country" name="country_box_ajax" value="1" ' . $checked . '  />';
+}
+
+/* Callback for a custom setting */
+function show_checkbox_address_ajax() {
+    $options = get_option( 'show_ajax_settings' );
+    $checked = '';
+    if( $options["show_address"] == 1 ) $checked = 'checked';
```

Comment:
there is a WP function called `checked`, but we can accept that with the first round

Reply:
ping about this one, as a reminder